components/button/README.md

```diff
@@ -1,28 +1,50 @@
 # Buttons
 
-## d2l-button
-
-The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
-
-![Button](./screenshots/button.png?raw=true)
+A Button is used to communicate and perform an action.
 
 ```html
+<!-- docs: demo -->
 <script type="module">
   import '@brightspace-ui/core/components/button/button.js';
+  import '@brightspace-ui/core/components/button/button-subtle.js';
+  import '@brightspace-ui/core/components/button/button-icon.js';
 </script>
-<d2l-button>My Button</d2l-button>
+<d2l-button>Button</d2l-button>
+<d2l-button primary>Primary Button</d2l-button>
+<d2l-button-subtle text="Subtle Button" icon="tier1:gear"></d2l-button-subtle>
+<d2l-button-icon text="Icon Button" icon="tier1:gear"></d2l-button-icon>
 ```
 
-**Properties:**
+## Best Practices
+<!-- docs: start best practices -->
+<!-- docs: start dos -->
+* Use buttons to cause an action or launch a workflow
+* Keep button text short - see "Writing" guidelines
+<!-- docs: end dos -->
+
+<!-- docs: start donts -->
+* Don't use multiple Primary buttons - the primary action should always be clear and obvious
+* Avoid icon-only buttons with unfamiliar icons, only use these for familiar concepts
+* Don't use buttons for navigation, use a link instead
+* Don't open menus with buttons - use a dropdown instead
+<!-- docs: end donts -->
+<!-- docs: end best practices -->
+
+## Button [d2l-button]
+
+The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
 
-| Property | Type | Description |
-|--|--|--|
-| `description` | String | A description to be added to the `button` for accessibility |
-| `disabled` | Boolean | Disables the button |
-| `disabledTooltip` | String | Tooltip text when disabled |
-| `primary` | Boolean | Styles the button as a primary button |
+```html
+<!-- docs: live demo
```

Comment:
I haven't been following along with the Daylight site PRs and how the parsing works, so this may be more difficult than it's worth.  But I assume if we moved this comment above the ` ```html`, then it wouldn't be displayed in the readme at all, like the other docs comments throughout.  Would that really mess up the parser though?

Reply:
It would, yeah. Text/comments vs code blocks are parsed out as separate blocks so it would complicate the parsing for sure. We could probably replace the comment with the opening tag for the demo component but it's a bit complicated by the fact that we don't treat the content strictly as a slot. I'll do some experimenting and see.

Reply:
Ok cool, yeah not trying to derail things too much.  Would be a nice-to-have, especially where we repeat the attributes already there in the `defaults` section